Replace all calls to strncpy with strlcpy, use strdup more, expose strlcat

CookiePLMonster opened the pull request:
# What's new

This PR improves the consistency of string copy calls, removing all uses of `strncpy` and `strcat`, at least from the code that isn't in submodules. `strlcpy` is preferable over `strncpy`, because it doesn't needlessly zero the remaining destination buffer, and it ensures null-termination.

Because `strlcat` is already used in mjs, I exposed it in the API. Please check if this is fine.

To make buffer sizes consistent, I modified the following constants to include the space for the null terminator. Please check if this is fine:
* `IBUTTON_KEY_NAME_SIZE`
* `INFRARED_MAX_BUTTON_NAME_LENGTH`
* `INFRARED_MAX_REMOTE_NAME_LENGTH`
* `MAX_TEXT_INPUT_LEN`

# Verification 

* Run a full suite of tests.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

skotopes commented on applications/debug/rpc_debug_app/rpc_debug_app.c:
```diff
@@ -60,7 +60,10 @@ static void rpc_debug_app_rpc_command_callback(const RpcAppSystemEvent* event, v
+            event->data.bytes.ptr,
+            event->data.bytes.size,
+            app->text_store,
+            sizeof(app->text_store));
```
There is one thing that I don't like about sizeof: compiler will not inform you if you change char[] with char*.

CookiePLMonster replied:
Fair point. Should I change all the occurrences back? Or does `COUNT_OF` account for it? AFAIK it doesn't.

skotopes replied:
yep, roll em back please

skotopes commented:
Un-draft when ready

CookiePLMonster committed:
Replace all calls to strncpy with strlcpy, use strdup more, expose strlcat

strlcpy doesn't zero the buffer and ensures null termination,
just like snprintf

strlcat is already used by mjs and it's a safe alternative to strcat,
so it should be OK to expose to apps
